diff --git a/node_modules/cz-conventional-changelog/engine.js b/node_modules/cz-conventional-changelog/engine.js
index 24a42b7..fab5ea5 100644
--- a/node_modules/cz-conventional-changelog/engine.js
+++ b/node_modules/cz-conventional-changelog/engine.js
@@ -76,16 +76,11 @@ module.exports = function(options) {
           default: options.defaultType
         },
         {
-          type: 'input',
+          type: 'list',
           name: 'scope',
-          message:
-            'What is the scope of this change (e.g. component or file name): (press enter to skip)',
-          default: options.defaultScope,
-          filter: function(value) {
-            return options.disableScopeLowerCase
-              ? value.trim()
-              : value.trim().toLowerCase();
-          }
+          message: 'Select the scope of this change:',
+          choices: globalThis.scopes.map(value => ({ label: value, value })),
+          default: globalThis.defaultScope
         },
         {
           type: 'input',
diff --git a/node_modules/cz-conventional-changelog/index.js b/node_modules/cz-conventional-changelog/index.js
index 7a4586d..16703c3 100644
--- a/node_modules/cz-conventional-changelog/index.js
+++ b/node_modules/cz-conventional-changelog/index.js
@@ -47,4 +47,8 @@ var options = {
   } catch (err) {}
 })(options);
 
-module.exports = engine(options);
+var engineInstance = engine(options);
+
+module.exports = engineInstance;
+module.exports.types = Object.keys(options.types);
+module.exports.prompter = engineInstance.prompter;
